hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AsyncServerSocket.java

```diff
@@ -144,6 +155,14 @@ public final SocketAddress getLocalAddress() {
 
     public abstract void listen(int backlog);
 
+    /**
+     * Accept incoming AsyncSocket.
+     * <p/>
+     * This call can be made from any thread, but the actual processing will happen on the eventloop-thread.
+     *
+     * @param consumer a function that is called when a socket has connected.
+     * @throws NullPointerException if consumer is null.
+     */
     public abstract void accept(Consumer<AsyncSocket> consumer);
```

Comment:
You may add `@Nonnull` annotation on consumer argument, as a companion to the `@throws`.

Reply:
I think such annotations are just noise.